pandas/core/window.py

```diff
@@ -996,7 +997,10 @@ def cov(self, other=None, pairwise=None, ddof=1, **kwargs):
             # only default unset
             pairwise = True if pairwise is None else pairwise
         other = self._shallow_copy(other)
-        window = self._get_window(other)
+        if self.is_freq_type:
```

Comment:
I tried changing `_get_window`, but a lot of methods (such as `var`, `median`, and `sum`) require an integer window and seem to be working fine right now. If this issue is not a problem in methods other than `cov`, changing the code only for `cov` seems to make more sense.